packages/modular-station/src/api.ts

```diff
@@ -143,34 +154,62 @@ export function withApi<
+		// a randomly generated hook returning the original value along with the symbolic marker. The hook name
+		// is randomly generated at runtime to avoid it being used by people that mess with internals of things (like we are doing here).
+		// The return value includes the symbolic marker to allow detecting whether the integration is the intended one
+		// even across duplicated installations, while the integration object is returned to recover access
+		// to the complete API after it is deleted by Astro.
+		Object.assign(integration.hooks, {
+			[API_RECOVER_HOOK_MARKER]: () => ({ integrationSymbol, integration }),
+		});
+		debug('Associated integration symbol with factory:', integration);
+		return integration;
 	};
 
 	const api: IntegrationApi<TOptions, TApi> = {
-		is: (integration: any): integration is AstroIntegration & TApi =>
+		is: (integration: any): integration is AstroIntegration & TApi => {
+			debug('Checking integration for symbol', { integrationSymbol, integration });
+			const recovered = integration?.hooks?.[API_RECOVER_HOOK_MARKER]?.();
+
 			// `integrationSymbol` is a non-exposed unique symbol, so it can only be present where
 			// the wrapper function above included it.
-			// Therefore an object matching this must be an integration constructed from that.
-			integration[integrationSymbol] === true,
-		fromIntegrations: (integrations) => integrations.find(api.is) ?? null,
+			// Therefore an object matching this must be an integration constructed from the wrapper.
+			return recovered?.integrationSymbol === integrationSymbol;
+		},
+		fromIntegration: (integration: AstroIntegration): (AstroIntegration & TApi) | null => {
+			debug('Checking integration for symbol', { integrationSymbol, integration });
+			const recovered = (integration.hooks as any)[API_RECOVER_HOOK_MARKER]?.();
+
+			return recovered?.integrationSymbol === integrationSymbol ? recovered.integration : null;
+		},
+		fromIntegrations: (integrations) => {
+			for (const integration of integrations) {
+				const target = api.fromIntegration(integration);
+				if (target !== null) return target;
+			}
+
+			return null;
+		},
 		fromConfig: (config) => api.fromIntegrations(config.integrations),
-		fromSetup: ({ config, updateConfig }, ...args) => {
+		fromSetup: ({ config }, ...args) => {
 			let instance = api.fromConfig(config);
 
 			if (instance === null) {
 				// If there is no instance of the integration currently installed, instantiate one
 				// using the given options and install it.
 				const fullIntegration = wrapper(...args);
-				updateConfig({
-					integrations: [fullIntegration],
-				});
-
-				// Also add the integration to the current configuration in case the API is used
-				// twice in the same consumer integration under different names.
+				debug(`Could not find integration "${fullIntegration.name}" in the config. Installing it.`);
 				config.integrations.push(fullIntegration);
```

Comment:
Unclear here whether the most appropriate is to just change the direct config value, just call `updateConfig` or both.

Quickly skimming through the code it seems like we should just change the given config object, but I'm not sure of it yet.

Reply:
My assumption is that it should be both, but I am not sure 🧙 🪄

Reply:
Checked the code in core and updated the code here appropriately